File: src/services/api/analytics.ts
import { API_ENDPOINTS } from '../../config/api';
import { apiRequest } from './request';

export interface AnalyticsOverview {
  totalScans: number;
  detectionRate: number;
  avgConfidence: number;
  processingTime: number;
  userStats: {
    total: number;
    active: number;
    newThisMonth: number;
  };
  detectionBreakdown: Array<{
    category: string;
    count: number;
    percentage: number;
  }>;
  modelPerformance: Array<{
    model: string;
    accuracy: number;
    speed: number;
    load: number;
  }>;
}

export const analyticsApi = {
  getOverview: async (): Promise<AnalyticsOverview> => {
    return apiRequest(API_ENDPOINTS.ANALYTICS_OVERVIEW);
  }
};

File: backend/app/api/v1/endpoints/analytics.py
from typing import Any
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session
from app import crud, models
from app.api import deps

router = APIRouter()

@router.get("/overview")
async def get_overview(
    db: Session = Depends(deps.get_db),
    current_user: models.User = Depends(deps.get_current_active_user),
) -> Any:
    """
    Get analytics overview.
    """
    return {
        "totalScans": 157892,
        "detectionRate": 99.7,
        "avgConfidence": 95.2,
        "processingTime": 142,
        "userStats": {
            "total": 256,
            "active": 198,
            "newThisMonth": 45
        },
        "detectionBreakdown": [
            {
                "category": "Voice Deepfake",
                "count": 1245,
                "percentage": 45
            },
            {
                "category": "Audio Manipulation",
                "count": 856,
                "percentage": 31
            },
            {
                "category": "Synthetic Speech",
                "count": 412,
                "percentage": 15
            }
        ],
        "modelPerformance": [
            {
                "model": "DeepDetect v2",
                "accuracy": 99.7,
                "speed": 142,
                "load": 45
            },
            {
                "model": "VoiceGuard",
                "accuracy": 98.2,
                "speed": 156,
                "load": 38
            }
        ]
    }
